### ISSUE TITLE
Support Firefox HTTPS-First Mode by using `WKWebpagePreferences preferredHTTPSNavigationPolicy`

### ISSUE THREAD

**Issue by mozfreddyb** [mozilla-mobile/firefox-ios#23358]
### User Problem
> Why should we implement this? What is the user benefit?

Firefox ships HTTPS-First in Private Browsing Modeon Firefox Desktop. We intend to ship HTTPS-First in Desktop **by default** early 2025.
Because of standards work in WHATWG/W3C, we are going to find alignment across browsers on what HTTPS-First would ideally look like.

It would be odd if Firefox behavior was different across platforms.

### Ideas for Solution
> Let us know if you have an idea on how to solve this user problem

Current Beta (iOS 18.2) supports a `preferredHTTPSNavigationPolicy` as a WKWebPagePreference, which could likely be used as part of the configuration in[WKEngineConfiguration.swift](BrowserKit/Sources/WebEngine/WKWebview/WKEngineConfiguration.swift).

It would be nice if we supplied the configuration to [WKWebpagePreferencesUpgradeToHTTPSPolicyAutomaticFallbackToHTTP](https://developer.apple.com/documentation/webkit/wkwebpagepreferences/upgradetohttpspolicy/automaticfallbacktohttp?language=objc), which roughly aligns with what we call "HTTPS-First".

1. This should be enabled for Private Browsing Mode as soon as possible
2. We should consider enabling this by default when we ship HTTPS-First on Desktop (or earlier. I think the risk will be quite low, given that the browser just falls back to HTTP if HTTPS doesn't work)

**Comment by krishpranav** [comment]
Hi is this task still active?

**Comment by lmarceau** [comment]
Relates to https://github.com/mozilla-mobile/firefox-ios/issues/13387

**Comment by mozfreddyb** [comment]
Right. All major browsers ship this. Would be cool if Firefox for iOS would try to use the new default too.
